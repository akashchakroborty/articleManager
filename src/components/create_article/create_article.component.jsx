import React, { useState, useEffect } from "react";
import { v4 as uuid } from "uuid";
import { connect } from "react-redux";
import getCurrentData from "../../utils/getCurrentData";
import getReadTimeInMin from "../../utils/getReadTimeInMin";
import { withRouter } from "react-router-dom";
import { createArticle } from "../../redux/articles/articles.action";
import FormComponent from "../form/form.component";

export const CreateArticle = ({ createArticle, history }) => {
  const [article, setarticle] = useState({
    title: "",
    claps: 0,
    description: "",
    id: uuid(),
    readTime: "0 min read",
    date: getCurrentData(),
    author: "",
    content: "",
    img: `https://loremflickr.com/240/400/programming?lock=${Math.floor(
      Math.random() * 1000
    )}`,
    avatar: `https://loremflickr.com/100/100/profilepic?lock=${Math.floor(
      Math.random() * 1000
    )}`,
  });

  const { title, description, author, content } = article;

  useEffect(() => {
    const articleReadTime = getReadTimeInMin(content);

    setarticle({ ...article, readTime: articleReadTime });
  }, [getReadTimeInMin(content)]);

  const handleSubmit = (event) => {
    event.preventDefault();
    console.log("Submitted");
    createArticle(article);
    history.push("/");
  };

  const handleChange = (event) => {
    event.preventDefault();
    const { value, name } = event.target;
    setarticle({ ...article, [name]: value });
  };

  return (
    <FormComponent
      handleSubmit={handleSubmit}
      title={title}
      description={description}
      author={author}
      content={content}
      handleChange={handleChange}
    />
  );
};

export default withRouter(connect(null, { createArticle })(CreateArticle));
