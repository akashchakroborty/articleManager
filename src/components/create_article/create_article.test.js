import React from "react";
import { mount } from "enzyme";
import { CreateArticle } from "./create_article.component";

describe("CreateArticle component", () => {
  let wrapper;
  let mockCreateArticle;
  let mockHistory;

  beforeEach(() => {
    mockCreateArticle = jest.fn();
    mockHistory = {
      push: jest.fn(),
    };

    const mockProps = {
      createArticle: mockCreateArticle,
      history: mockHistory,
    };

    wrapper = mount(<CreateArticle {...mockProps} />);
  });

  it("should render CreateArticle component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange method when FormComponent changes", () => {
    const value = "hey";
    wrapper
      .find("input")
      .at(0)
      .simulate("change", {
        target: { value, name: "title" },
      });
    expect(wrapper.find("input").at(0).instance().value).toEqual(value);
  });

  it("should call handleSubmit method when FormComponent is submitted", () => {
    const logSpy = jest.spyOn(console, "log");
    wrapper.find("form").at(0).simulate("submit");
    expect(logSpy).toHaveBeenCalledWith("Submitted");
  });
});
