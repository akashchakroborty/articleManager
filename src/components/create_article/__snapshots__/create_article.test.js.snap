// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateArticle component should render CreateArticle component 1`] = `ReactWrapper {}`;
